### PR TITLE
fix(auth): silent renew auth token between refreshes

### LINKED ISSUE #695: [Auth] Silent renew auth token if expired in between page loads
We were seeing testers during bug bashes hit 403s when sitting on the upload page for more than 10 minutes. This is a hotfix meant to fix this bug for the beta / next bug bash on Friday, but the react-oidc-context library will be revisited after launch. A lot of people have this issue it seems when users idle on a page, with the most common approach being to pop up an alert or use the approach I do in this PR: silently renewing the tokens if they expire. I'd rather lean on just `automaticSilentRenew`, and I think it may work in later versions of the library if we upgraded, but I'd like to fix it for now for the next bug bash then do a more thorough look at the `automaticSilentRenew` behavior when we get more time.

The obvious problem with this implementation is that currently this causes a re-render when a user's token expires (if they sit on the same page for 10 minutes). I think it's probably a mix of the useEffect but also maybe our config generally. I'd like to dig in a little deeper, but for now, let's fix it and then go back and try upgrading / creating a more nuanced auth strategy (maybe even upload page specific) after the next bug bash / post-mvp. I'm also going to ask that we bump the timeout to 15 minutes, and I think this issue will be mostly solved in the short-term. Here's [the stub of a ticket](https://github.com/cfpb/sbl-frontend/issues/696) for post-mvp work, and I might explore a more refined solution just for the upload page later.

## Current behavior
- If I sit on a page for 10 minutes, my token expires unless I refresh the page.

## Expected behavior
- If I sit on a page for 10 minutes, my token expires, the token refreshes, I can still perform actions.

## Steps to replicate behavior (include URLs)

1. Sit on the upload page for 15 minutes then try to upload.

### PR DESCRIPTION
**Pull request by billhimmelsbach**
We were seeing testers during bug bashes hit 403s when sitting on the upload page for more than 10 minutes. This is a hotfix meant to fix this bug for the beta / next bug bash on Friday, but the react-oidc-context library will be revisited after launch. A lot of people have this issue it seems when users idle on a page, with the most common approach being to pop up an alert or use the approach I do in this PR: silently renewing the tokens if they expire. I'd rather lean on just automaticSilentRenew, and I think it may work in later versions of the library if we upgraded, but I'd like to fix it for now for the next bug bash then do a more thorough look at the automaticSilentRenew behavior when we get more time.

The obvious problem with this implementation is that currently this causes a re-render when a user's token expires (if they sit on the same page for 10 minutes). I think it's probably a mix of the useEffect but also maybe our config generally. I'd like to dig in a little deeper, but for now, let's fix it and then go back and try upgrading / creating a more nuanced auth strategy (maybe even upload page specific) after the next bug bash / post-mvp. I'm also going to ask that we bump the timeout to 15 minutes, and I think this issue will be mostly solved in the short-term. Here's https://github.com/cfpb/sbl-frontend/issues/696 for post-mvp work, and I might explore a more refined solution just for the upload page later.

Closes #695 

## Changes

- adds a silent renewal of our auth tokens if they expire between page loads

## How to test this PR

1. Sit on the upload page for 15 minutes, your token expires, but the token refreshes and you can still perform actions.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -109,6 +110,18 @@ function BasicLayout(): ReactElement {
   const headerLinks = [...useHeaderAuthLinks()];
   const location = useLocation();
 
+  const auth = useSblAuth();
+  const navigate = useNavigate();
```
Comment (meissadia) [inline src/App.tsx:114]: Unused

**Review by meissadia — CHANGES_REQUESTED** [review]
Silent refresh seems to be working well, great addition. Just needs a bit of code cleanup.

File: src/App.tsx (hunk 1 above)
Comment (billhimmelsbach) [reply src/App.tsx:114]: Good catch @meissadia. I had bigger dreams for error handling here, but for a future #696. (also my after save actions for cleanup stuff are dead still on the new version of VS Code 😔)

Comment (billhimmelsbach) [reply src/App.tsx]: [Updated](https://github.com/cfpb/sbl-frontend/pull/697/commits/40988b22e9b3b34b485f6b0caba5c166a96ba393) @meissadia!

File: src/App.tsx [hunk 2]
```diff
@@ -22,14 +22,15 @@ import FilingWarnings from 'pages/Filing/FilingApp/FilingWarnings';
...
-import { Suspense, lazy } from 'react';
+import { Suspense, lazy, useEffect } from 'react';
 import {
   BrowserRouter,
   Navigate,
   Outlet,
   Route,
   Routes,
   useLocation,
+  useNavigate,
```
Comment (meissadia) [inline src/App.tsx:33]:
```suggestion
```
Also unused.

**Review by meissadia — DISMISSED** [review]

File: src/App.tsx (hunk 2 above)
Comment (billhimmelsbach) [reply src/App.tsx:33]: Gonna be a real sad time without VSCode linting...

Comment (billhimmelsbach) [reply src/App.tsx:33]: Found a "solution" to the linting issues for now.

**Review by meissadia — APPROVED** [review]

**Pull request merged**